sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleServer.scala

```diff
@@ -343,52 +343,59 @@ class RapidsShuffleServer(transport: RapidsShuffleTransport,
-                              logError(s"Error while handling TransferResponse: " +
-                                s"${transferResponseTx.getErrorMessage}")
-                            case _ =>
+              bufferTx.getStatus match {
+                case TransactionStatus.Success =>
+                  logDebug(s"Done with the send for ${bufferSendState} with ${buffersToSend}")
+
+                  if (bufferSendState.hasNext) {
+                    // continue issuing sends.
+                    logDebug(s"Buffer send state ${bufferSendState} is NOT done. " +
+                      s"Still pending: ${pendingTransfersQueue.size}.")
+                    bssExec.synchronized {
+                      bssContinueQueue.add(bufferSendState)
+                      bssExec.notifyAll()
+                    }
+                  } else {
+                    val transferResponse = bufferSendState.getTransferResponse()
+
+                    val requestTx = bufferSendState.getRequestTransaction
+                    logDebug(s"Handling transfer request $requestTx for executor " +
+                      s"$peerExecutorId with $buffersToSend")
+
+                    // send the transfer response
+                    requestTx.respond(transferResponse.acquire(),
+                      transferResponseTx => {
+                        withResource(transferResponseTx) { _ =>
+                          withResource(transferResponse) { _ =>
+                            transferResponseTx.getStatus match {
+                              case TransactionStatus.Cancelled | TransactionStatus.Error =>
+                                logError(s"Error while handling TransferResponse: " +
+                                  s"${transferResponseTx.getErrorMessage}")
+                              case _ =>
+                            }
                           }
                         }
-                      }
-                    })
-
-                  // wake up the bssExec since bounce buffers became available
-                  logDebug(s"Buffer send state ${buffersToSend} is done. Closing. " +
-                    s"Still pending: ${pendingTransfersQueue.size}.")
+                      })
+
+                    // wake up the bssExec since bounce buffers became available
+                    logDebug(s"Buffer send state ${buffersToSend} is done. Closing. " +
+                      s"Still pending: ${pendingTransfersQueue.size}.")
+                    bssExec.synchronized {
+                      bufferSendState.close()
+                      bssExec.notifyAll()
+                    }
+                  }
+                case _ =>
+                  // errored or cancelled
+                  logError(s"Error while sending buffers $bufferTx.")
                   bssExec.synchronized {
                     bufferSendState.close()
                     bssExec.notifyAll()
                   }
                 }
-              }
```

Comment:
Nit: Looks like the wrong closing brace was removed here, as the brace above is mis-indented.